MapboxNavigation/RouteMapViewController.swift

```diff
@@ -808,18 +809,56 @@ extension RouteMapViewController: NavigationViewDelegate {
                     } else {
                         currentName = nil
                     }
+                    
+                    if let line = feature as? MGLPolylineFeature {
+                        if let text = line.attribute(forKey: "ref") as? String, let shieldRawValue = line.attribute(forKey: "shield") as? String, let reflen = line.attribute(forKey: "reflen") {
+                            
+                            let currentShield = HighwayShield.RoadType(rawValue: shieldRawValue)
+                            let textColor = currentShield?.textColor ?? .black
+                            
+                            let imageName = "\(shieldRawValue)-\(reflen)"
+                            if let image = mapView.style?.image(forName: imageName) {
+                                currentShieldName = attributedString(withFont: UIFont.boldSystemFont(ofSize: 12.0), shieldImage: image, text: text, color: textColor)
+                            }
+                        }
+                    }
+                    
                 }
             }
         }
         
-        if smallestLabelDistance < 5 && currentName != nil {
-            navigationView.wayNameView.text = currentName
+        let hasWayName = currentName != nil || currentShieldName != nil
+        if smallestLabelDistance < 5 && hasWayName  {
+            if let currentShieldName = currentShieldName {
+                navigationView.wayNameView.attributedText = currentShieldName
+            } else if let currentName = currentName {
+                navigationView.wayNameView.text = currentName
+            }
             navigationView.wayNameView.isHidden = false
         } else {
             navigationView.wayNameView.isHidden = true
         }
     }
     
+    private func attributedString(withFont font: UIFont, shieldImage: UIImage, text: String, color: UIColor?) -> NSAttributedString {
+        let attachment = RoadNameLabelAttachment()
+        attachment.font = font
+        attachment.text = text
```

Comment:
What are we getting out of `RoadNameLabelAttachment` over `NSAttributedString` directly? Having a new text prop sounds a little funny when there is already an initializer for string.

Reply:
`RoadNameLabelAttachment` should handle the text and image composition. Our primary goal was to use the text prop as the accessibility text/hint for that road name label attachment image. However, it has been revised to allow the `RoadNameLabelAttachment` to handle the task of inserting the text in the image.